heron/api/src/java/com/twitter/heron/api/Config.java

```diff
@@ -147,6 +147,31 @@
    */
   public static final String TOPOLOGY_COMPONENT_RAMMAP = "topology.component.rammap";
   /**
+   * Is this topology stateful? The format of this flag is boolean
+   */
+  public static final String TOPOLOGY_STATEFUL = "topology.stateful";
```

Comment:
Agree it should be a topology specific config:
- if people want a component to be stateless, they can simply not to store any states, even with this config is set true
- it does not make sense to make this config per component; the whole topology DAG is a computing job and the global state will be correct only when all components have set `topology.component.state.enabled` true.